File: Agora-RTC-With-Voice-Changer-iOS-Objective-C/Agora-RTC-With-Voice-Changer-iOS-Objective-C/Cell/EffectCell.m
//
//  EffectCell.m
//  Agora-RTC-With-Voice-Changer-iOS-Objective-C
//
//  Created by ZhangJi on 2018/5/4.
//  Copyright © 2018 ZhangJi. All rights reserved.
//

#import "EffectCell.h"

@implementation EffectCell

- (void)awakeFromNib {
    [super awakeFromNib];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

}

- (IBAction)doValueChanged:(UISlider *)sender {
    self.valueLabel.text = [NSString stringWithFormat:@"%d", (int)sender.value];
    if ([self.delegate respondsToSelector:@selector(effectCell:sliderValue:didChangedAtIndexPath:)]) {
        [self.delegate effectCell:self sliderValue:sender.value didChangedAtIndexPath:self.indexPath];
    }
}

- (CSRange)rangeOfBandFrequency:(AgoraAudioEqualizationBandFrequency)bandFrequency {
    return (CSRange){-15.0, 15.0};
}

- (CSRange)rangeOfReverType:(AgoraAudioReverbType)reverType {
    switch (reverType) {
        case AgoraAudioReverbDryLevel: return (CSRange){-20.0, 10.0};
        case AgoraAudioReverbWetLevel: return (CSRange){-20.0, 10.0};
        case AgoraAudioReverbRoomSize: return (CSRange){0.0, 100.0};
        case AgoraAudioReverbStrength: return (CSRange){0.0, 200.0};
        case AgoraAudioReverbWetDelay: return (CSRange){0.0, 100.0};
    }
}


@end

File: iOS&macOS/Agora-RTC-With-Voice-Changer-iOS/Agora-RTC-With-Voice-Changer-iOS/EffectCell.swift
//
//  EffectCell.swift
//  Agora-RTC-With-Voice-Changer-iOS
//
//  Created by CavanSu on 14/03/2018.
//  Copyright © 2018 Agora. All rights reserved.
//

import UIKit

protocol EffectCellDelegate: NSObjectProtocol {
    func cell(_ cell: EffectCell, didSliderValueChange indexPath: IndexPath, value: Double)
}

class EffectCell: UITableViewCell {
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var valueLabel: UILabel!
    @IBOutlet weak var valueSlider: UISlider!
    
    weak var delegate: EffectCellDelegate?
    
    var value: Double = 0.0
    var indexPath: IndexPath!
    
    var title: String! {
        didSet {
            titleLabel.text = title
        }
    }
    
    @IBAction func doValueChanged(_ sender: UISlider) {
        value = Double(sender.value)
        
        if indexPath.section == 0 {
            valueLabel.text = NSString(format: "%.1f", value) as String
        }
        else {
            valueLabel.text = "\(Int(value))"
        }
        
        delegate?.cell(self, didSliderValueChange: indexPath, value: value)
    }
}
